scripts/stetho_open.py

```diff
@@ -121,7 +121,7 @@ def select_service(self, service):
   def _read_exactly(self, num_bytes):
     buf = b''
     while len(buf) < num_bytes:
-      new_buf = self.sock.recv(num_bytes)
+      new_buf = self.sock.recv(num_bytes - len(buf))
```

Comment:
Testing fail.  Also, prefer that you expose this function for `dumpapp` to share in common and fix the new_buf being empty bug here as well.  The correct response is something like `raise IOError('EOF')`